libsolidity/interface/StandardCompiler.cpp

```diff
@@ -576,12 +578,13 @@ Json::Value StandardCompiler::compileInternal(Json::Value const& _input)
 	}
 	m_compilerStack.setLibraries(libraries);
 
-	Json::Value metadataSettings = settings.get("metadata", Json::Value());
-
-	if (auto result = checkMetadataKeys(metadataSettings))
-		return *result;
-
-	m_compilerStack.useMetadataLiteralSources(metadataSettings.get("useLiteralContent", Json::Value(false)).asBool());
+	if (settings.isMember("metadata"))
```

Comment:
The style of the other lines suggests that this check is not needed, or is it?

Reply:
I had the same question.

Reply:
Optional objects are using this check. For example, `if (settings.isMember("optimizer"))` is present on L514 so I added this for consistency. That being said, it was working earlier as well when `metadata` was absent so we probably don't need the check.

Reply:
Please remove the check

Reply:
Removed.